### ISSUE TITLE
Unable to open the application using Normal port for Istio-gateway using Metallb for RKE Cluster

### ISSUE THREAD

**Issue by siddharth25pandey** [metallb/metallb#1909]
I'm using Metallb for provisioning the Load Balancer in RKE cluster. Then I installed Istio for serivce mesh. Inside that, Istio Gateway is only allowing the random NodePort of the Istio-ingress gateway service to open the application after the provisioning of load balancer, why the normal port mentioned in the values.yaml inside the Istio-Gateway is not accessible to open the application.
[Istio Pods & Services](https://i.stack.imgur.com/4h705.png)
Here, I'm able to open the application through 31940 port, but unable to open the application by using port 80(http) & 443 (https).

**Comment by rniranjan89** [comment]
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
name: first-pool
namespace: metallb-system
spec:
addresses:

192.168.1.240-192.168.1.250
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
name: example
namespace: metallb-system

@siddharth25pandey  I hope you applied both IPAddressPool and L2Advertisement?

**Comment by siddharth25pandey** [comment]
@rniranjan89, Yeah I applied both IPAddressPool and L2Advertisement.
`apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
  name: first-pool
  namespace: metallb-system
spec:
  addresses:
  - 3.218.177.110/32 
`
`apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: example
  namespace: metallb-system
spec:
  ipAddressPools:
  - first-pool
`
Can you please help

**Comment by rniranjan89** [comment]
@siddharth25pandey can you send me more details about your cluster, RKE or RKE2? which version network? istio version .. etc... , and also is it accessible from inside the cluster?

**Comment by siddharth25pandey** [comment]
> @siddharth25pandey can you send me more details about your cluster, RKE or RKE2? which version network? istio version .. etc... , and also is it accessible from inside the cluster?

Sure @rniranjan89 , I'm using RKE version 1.4.2 and Istio version, 1.17.2 (base, Istiod & gateway all through helm separately), networking.istio.io/v1alpha3. Yes, using 31940 port is publicly accessible (withing as well as outiside cluster). Currently I have a one single node RKE cluster (which have all 3 controleplane, etcd & worker in the same node (EC2 instance))

**Comment by rniranjan89** [comment]
@siddharth25pandey   we  have ingress gateway as Load balancer with external ip (x.x.x.x) in istio-system namespace with 80 and 443 ports open, after that you have Gateway which has port 80 and 443 opened for a particular domain name /host  and virtual service is connects with gateway to pass it to your application port, this is the flow

**Comment by rniranjan89** [comment]
@siddharth25pandey below is the troubleshooting guide for Metallb, can you Curl or ping the load balancer ip inside the cluster and see if you are able to access your application, if you can access it then it is definitely issue with your L2Advertisement and IPAddressPool 

https://metallb.universe.tf/configuration/troubleshooting/

**Comment by siddharth25pandey** [comment]
@rniranjan89 After doing, `kubectl -n istio-system get endpoints istio-gateway`, it showed the private ip with ports as endpoints
`10.42.0.23:15021,10.42.0.23:8080,10.42.0.23:8443`, Able to curl this (10.42.0.23:8080) inside the cluster, as well as other routes as defined in the gateway file. But through the public ip (3.218.177.110) Able to successfully curl without mentioning any port.
every route is working (3.218.177.110, 3.218.177.110/new) inside the cluster, after curling it!

**Comment by siddharth25pandey** [comment]
> @siddharth25pandey you will have ingress gateway as Load balancer with external ip (x.x.x.x) in istio-system namespace with 80 and 443 ports open, after that you will have Gateway which has port 80 and 443 opened for a particular domain name /host and virtual service connects with gateway to pass it to your application port, this is the flow

@rniranjan89 I think the flow is correct & implemented the same, ports are open

**Comment by siddharth25pandey** [comment]
As of now, after curl it's working perfectly inside the cluster, but if hitting from any other server outside the RKE cluster, it's only accessible through a specific port!, i.e the random NodePort allocation of Istio-ingress gateway service

**Comment by siddharth25pandey** [comment]
> Did you follow the troubleshooting guide? Does arping work? If so, does the traffic reach the right node when you curl (run tcpdump on the node to see if it's getting through).

@fedepaol Yes, I followed the troubleshoot guide, I have one node cluster, inside that after curling it through public ip, it's working perfectly inside the cluster, but if hitting from any other server outside the RKE cluster, it's only accessible through a specific port!

**Comment by fedepaol** [comment]
> > Did you follow the troubleshooting guide? Does arping work? If so, does the traffic reach the right node when you curl (run tcpdump on the node to see if it's getting through).
> 
> @fedepaol Yes, I followed the troubleshoot guide, I have one node cluster, inside that after curling it through public ip, it's working perfectly inside the cluster, but if hitting from any other server outside the RKE cluster, it's only accessible through a specific port!

Does arping work from outside?
If you curl from outside and tcpdump from the node, do you see the traffic getting to the node?

**Comment by siddharth25pandey** [comment]
> > > Did you follow the troubleshooting guide? Does arping work? If so, does the traffic reach the right node when you curl (run tcpdump on the node to see if it's getting through).
> > 
> > 
> > @fedepaol Yes, I followed the troubleshoot guide, I have one node cluster, inside that after curling it through public ip, it's working perfectly inside the cluster, but if hitting from any other server outside the RKE cluster, it's only accessible through a specific port!
> 
> Does arping work from outside? If you curl from outside and tcpdump from the node, do you see the traffic getting to the node?

If I'll curl from outside, it's not working

**Comment by siddharth25pandey** [comment]
> @siddharth25pandey did you manage to solve this? If not, please try to follow the troubleshooting guide https://metallb.universe.tf/troubleshooting/

@fedepaol Unable to resolve yet!

**Comment by fedepaol** [comment]
DId you follow the guide? Does arp work?

**Comment by siddharth25pandey** [comment]
> DId you follow the guide? Does arp work?

@fedepaol Yes, I followed the doc & arp doesn't worked sadly

**Comment by fedepaol** [comment]
What step doesn't work?
